sycl/doc/extensions/experimental/sycl_ext_oneapi_reduction_properties.asciidoc

```diff
@@ -124,14 +125,17 @@ template <typename T, typename BinaryOperation, typename PropertyList>
 __unspecified__ reduction(T* var, const T& identity, BinaryOperation combiner,
                           PropertyList properties);
 
-template <typename T, typename Extent, typename BinaryOperation, typename PropertyList>
+template <typename T, size_t Extent, typename BinaryOperation, typename PropertyList>
 __unspecified__ reduction(span<T, Extent> vars, const T& identity,
                           BinaryOperation combiner,
                           PropertyList properties);
 
 }
 ----
 
+_Mandates_: The specified `properties` are compatible with the reduction,
+according to the description of each individual property.
```

Comment:
I think this should be a _Constraint_.  This is how I've been specifying other functions that take a property list.  For example:

https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/experimental/sycl_ext_oneapi_kernel_compiler.asciidoc#new-free-functions-to-create-and-build-kernel-bundles

> _Constraints:_ Available only when `PropertyListT` is an instance of `sycl::ext::oneapi::experimental::properties` which contains no properties other than those listed below in the section "New properties".

I also think we could add a _Mandates_ clause like:

> The `reduction` functions that take no `identity` parameter have the following clause:
>
> _Mandates:_ If `PropertyList` contains the `initialize_to_identity` property, then the identity of `BinaryOperation` operation must be identifiable via the `known_identity` trait class.

Reply:
Oh, my intent here was to capture the same as your _Mandates_ -- I'd completely overlooked the fact we should constrain the property list itself.

I like your wording, so I'll make these changes tomorrow.

Reply:
Ok, wording is fixed in f9185fa57fd719ffcf0721283c4e334ad9d656d7.